### PR TITLE
simple api attempt 1

### PR DESCRIPTION
**Pull request by guzba**
Question this raised: should the image width and height be considered bounds for wrapping the text? eg:

```nim
for path in font.typeset(text):
```
into
```nim
for path in font.typeset(text, bounds = image.wh):
```

or should it just draw a single line of text and if you want wrapping etc use the typeset api directly?

### CODE REVIEW

**Comment by treeform** [comment]
Skia/Cairo use a different bounds API. I don't think image size should be the bounds as usually you are drawing text some place random on an image.

File: src/pixie.nim [hunk 1]
```diff
@@ -317,3 +317,46 @@ proc strokePolygon*(
...
+  transform: Vec2 | Mat3 = vec2(0, 0)
+) =
+  for path in font.typeset(text):
+    mask.fillPath(path, transform)
+
+proc strokeText*(
+  image: Image,
+  font: Font,
+  text: string,
+  pos = vec2(0, 0),
```
Comment (treeform) [inline src/pixie.nim:345]: remove, has both pos and transform.

File: src/pixie.nim [hunk 2]
```diff
@@ -317,3 +317,46 @@ proc strokePolygon*(
...
+  transform: Vec2 | Mat3 = vec2(0, 0)
+) =
+  for path in font.typeset(text):
+    image.fillPath(path, color, transform)
+
+proc fillText*(
+  mask: Mask,
+  font: Font,
+  text: string,
+  color: SomeColor,
```
Comment (treeform) [inline src/pixie.nim:335]: remove color, mask.fillText should not take color

File: src/pixie.nim [hunk 3]
```diff
@@ -317,3 +317,46 @@ proc strokePolygon*(
...
+  strokeWidth = 1.0
+) =
+  for path in font.typeset(text):
+    image.strokePath(path, color, transform, strokeWidth)
+
+proc strokeText*(
+  mask: Mask,
+  font: Font,
+  text: string,
+  color: SomeColor,
```
Comment (treeform) [inline src/pixie.nim:357]: remove color, mask.fillText should not take color

**Comment by treeform** [comment]
Having bounds, alignment and other optional text params be part of fill/strokeText might not be a bad idea.

File: src/pixie.nim (hunk 2 above)
Comment (guzba) [reply src/pixie.nim]: done

File: src/pixie.nim (hunk 3 above)
Comment (guzba) [reply src/pixie.nim]: done

File: src/pixie.nim (hunk 1 above)
Comment (guzba) [reply src/pixie.nim]: done

**Comment by guzba** [comment]
added a test for each api variant

File: tests/test_fonts.nim [hunk 4]
```diff
@@ -1,4 +1,34 @@
-import pixie, strformat
+import pixie, pixie/fileformats/png, strformat
+
+block:
+  let font = readFont("tests/fonts/Roboto-Regular.ttf")
+  font.size = 64
+  let image = newImage(200, 100)
+  image.fill(rgba(255, 255, 255, 255))
+  image.filLText(font, "fill", rgba(0, 0, 0, 255))
```
Comment (treeform) [inline tests/test_fonts.nim:8]: fillText not filLText

File: tests/test_fonts.nim [hunk 5]
```diff
@@ -1,4 +1,34 @@
...
+  let mask = newMask(200, 100)
+  mask.fillText(font, "fill")
+  writeFile("tests/fonts/mask_fill.png", mask.encodePng())
+
+block:
+  let font = readFont("tests/fonts/Roboto-Regular.ttf")
+  font.size = 64
+  let mask = newMask(200, 100)
+  mask.strokeText(font, "stroke")
+  writeFile("tests/fonts/mask_stroke.png", mask.encodePng())
```
[the remark this answers is not in the document]
Comment (guzba) [reply tests/test_fonts.nim:31]: added

File: tests/test_fonts.nim (hunk 4 above)
Comment (guzba) [reply tests/test_fonts.nim]: fixed

**Pull request merged**